### PR TITLE
Add "batch" requests submission to io-tester

### PR DESCRIPTION
**Pull request by xemul**
Currently io-tester supports two kinds of IO workloads -- continuous and rated. In the former case requests come one after another from several fibers, in the latter case fibers submit requests at a given rate one at a time.

In real life the incoming requests pattern is not that simple. A nice approach to pretty common pattern can be to submit requests in batches at some rate. Say, once every 200 milliseconds 10 requests are queued at once.

Of course, in reality it's even more convoluted, e.g. once every 200 ms there can comes a _sequence_ of, say, 10 requests with some small pauses in between, but submitting them in a one-time batch gives good approximation.

Also this PR simplifies the continuous and rate-based submission code to make the new batch parameter apply smoother.

### CODE REVIEW

**Comment by tchaikov** [comment]
nit, s/smother/smoother/, because smother has some bad meaning. but this change is an improvement.

File: apps/io_tester/io_tester.cc [hunk 1]
```diff
@@ -354,21 +354,11 @@ class class_data {
                     return do_until([this, stop] { return std::chrono::steady_clock::now() > stop || requests() > limit(); }, [this, buf, stop, pause, &intent, &in_flight] () mutable {
                         auto start = std::chrono::steady_clock::now();
                         in_flight++;
-                        return issue_request(buf, &intent).then_wrapped([this, start, pause, stop, &in_flight] (auto size_f) {
```
[the remark this answers is not in the document]
Comment (xemul) [reply apps/io_tester/io_tester.cc:357]: It's a separate branch in my queue, yes

File: apps/io_tester/io_tester.cc [hunk 2]
```diff
@@ -919,6 +926,9 @@ struct convert<shard_info> {
         if (node["rps"]) {
             sl.rps = node["rps"].as<unsigned>();
         }
+        if (node["batch"]) {
+            sl.batch = node["batch"].as<unsigned>();
+        }
```
[the remark this answers is not in the document]
Comment (xemul) [reply apps/io_tester/io_tester.cc:932]: Added

**Comment by xemul** [comment]
upd:
- updated doc/io-tester.md
- fixed PR comment wording

**Review by pwrobelse — APPROVED** [review]
Looks good to me :+1:

**Comment by xemul** [comment]
@avikivity , please consider merging

**Comment by xemul** [comment]
@avikivity , please consider merging

**Comment by xemul** [comment]
@avikivity , please consider merging

**Pull request merged**